### ISSUE TITLE
TypeNameHandling is unsafe still warns when a custom SerialisationBinder is defined

### ISSUE THREAD

**Issue by girlpunk** [returntocorp/sgrep-rules#2299]
**Describe the bug**
The `csharp.lang.security.insecure-deserialization.newtonsoft.insecure-newtonsoft-deserialization` rule reports that "a custom SerializationBinder whenever using a setting other than TypeNameHandling.None". However, the rule still shows a warning when a SerializationBinder is defined.

**To Reproduce**
```csharp
var serialiser = new JsonSerializer
{
    TypeNameHandling = TypeNameHandling.Auto,
    SerializationBinder = new KnownTypesBinder(),
};
```

**Expected behavior**
The rule should not warn in this situation.

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [X] P2: annoying but not blocking me

**Additional Context**
